src/datasets/parallel/parallel.py

```diff
@@ -0,0 +1,113 @@
+import contextlib
+from multiprocessing import Pool, RLock
+
+import joblib
```

Comment:
The import should be moved to `_map_with_joblib`, since joblib is an optional dependency